lib/extract-js.js

```diff
@@ -0,0 +1,216 @@
+var htmlparser = require('htmlparser2');
+var Errors = require('./errors');
+
+/**
+ * Html file representation (needs for errors output).
+ *
+ * @name HtmlFile
+ */
+var HtmlFile = function(filename, source) {
+    this._filename = filename;
+    this._lines = source.split(/\r\n|\r|\n/);
```

Comment:
@aik099 Line end is a `\r` (mac) or `\n` (unix) or `\r\n` (win). We can tune for gracefully parse mixes of them, but it really can be substituted with `/\r\n?|\n/`. Right?